frontend/src/components/SignIn.jsx

```diff
@@ -0,0 +1,60 @@
+import React, { useState } from 'react';
+
+const SignIn = () => {
+  const [email, setEmail] = useState('');
+  const [password, setPassword] = useState('');
+  const [username, setUsername] = useState('');
+  const [isSignUp, setIsSignUp] = useState(true); // Start with sign up mode
+
+  const handleSubmit = async (e) => {
+    e.preventDefault();
+    if (isSignUp) {
+      // Handle sign up
+      const response = await fetch(import.meta.env.VITE_BACKEND_URL + '/api/register', {
+        method: 'POST',
+        headers: {
+          'Content-Type': 'application/json',
+        },
+        body: JSON.stringify({ email, username, password }),
+      });
```

Comment:
Using some API client libraries like axios can help you to reduce the boilerplate and save some time on discovering how to handle errors in `fetch`, maybe some other edge cases.